Infinit sliding

HannaJeon opened the pull request:
- 무한 슬라이딩 기능 구현 완료

skim451 committed:
Fixed few bugs on autocomplete. Code share.

skim451 committed:
Added trim() for search word

Jeju-Boobby committed:
Trying infinite sliding

Jeju-Boobby committed:
Fixed conflicts

HannaJeon committed:
무한슬라이딩 css  준비

무한슬라이딩을 위한 prototype hover기능을 이용해서 작동되게 함.

skim451 committed:
code share

skim451 committed:
Added menu sliding feature.

밑반찬 slide is now slide with left right buttons.

crongro commented on src/script.js:
```diff
@@ -179,10 +191,67 @@ class EventHandler {
 
+class MenuSlider {
+    constructor(menuSlider, leftButton, rightButton) {
+        this.position = -980 * 3; 
```
MenuSlider를 좀더 범용적으로 동작시키려면 position값은 인자로 받도록 하면 좋아요.

HannaJeon replied:
완료

crongro commented on src/script.js:
```diff
@@ -179,10 +191,67 @@ class EventHandler {
+        this.leftButton = leftButton; 
+        this.rightButton = rightButton; 
+
+        menuSlider.addEventListener('transitionend', () => this.onTransitionEnd());
```
아래 코드와 동일한 코드. 불필요하게 함수하나를 더 감쌀필요는 없을 듯.

menuSlider.addEventListener('transitionend', this.onTransitionEnd);

HannaJeon replied:
완료

crongro commented on src/script.js:
```diff
@@ -179,10 +191,67 @@ class EventHandler {
+    }
+
+    onTransitionEnd() {
+        if(this.position > -980) {
```
위에서 말했지만 980은 인자로 받으면 될 값이네요. 980만봐서는 코드의의도를 알 수가 없음으로.

HannaJeon replied:
완료

crongro commented on src/script.js:
```diff
@@ -179,10 +191,67 @@ class EventHandler {
+            this.menuSlider.style["transform"] = `translate3d(${this.position}px, 0px, 0px)`;
+        }
+
+        this.leftButton.disabled = false;  
```
아래 두 줄 정도도 하나의 함수로 만들어서 그 함수를 부르는 게 좀더 나을 듯.

HannaJeon replied:
완료

crongro commented on src/script.js:
```diff
@@ -179,10 +191,67 @@ class EventHandler {
+
+    onRightButtonClick() {
+        this.rightButton.disabled = true; 
+        if(this.position <= -980 * 5) {
```
왜 5고 왜 2인지..
panel의 갯수때문에?

HannaJeon replied:
그렇습니다.. 주석달고 계산을 조금 더 명시적으로 변경하였습니다.

crongro commented on src/script.js:
```diff
@@ -179,10 +191,67 @@ class EventHandler {
+        } else {
+            this.position -= 980;
+        }
+        this.menuSlider.style["transition"] = 'transform 0.5s ease-in-out';
```
0.5s가 다른 시간으로 바뀌면??  이것도역시 slide 생성자에 선언하거나, 생성자의 인자로 받던가..

HannaJeon replied:
완료

crongro reviewed (changes requested):
slidemenu 컴포넌트의 생성자-메서드 사용이 적절해요.
크기도 좋고요.
코드안에 데이터를 좀더 분리해서 범용성을 올리면 좋겠어요.
그리고 menuslide에 대한 테스트코드 구현하시고요.

HannaJeon committed:
리뷰1 해결

HannaJeon committed:
HTML 줄이고 js panel계산 변경

HannaJeon committed:
transitionTime 인자로 받음

HannaJeon committed:
button disabled 함수화를 하였다

smartyjpark committed:
menuSlider 테스트코드 작성

skim451 committed:
Added auto sliding feature

The menu slide now slide every 2 secs.
The button click event resets the auto slide timer.

skim451 committed:
Delete redundant lines

skim451 committed:
Add stop feature for autoslide when mouseover

Mouse over event now turns off the auto slide and
mouse out restarts the auto slide.

crongro commented on src/script.js:
```diff
@@ -211,9 +211,22 @@ class MenuSlider {
+    }
+
+    stopAutoSlide() {
+        clearInterval(this.autoInterval); 
```
setInterval은 잘 안씁니다. 정기적으로 실행되지 않습니다. 비동기 로직이 다른 처리때문에 지연실행될 수 있어요.
